lib/puppet/provider/puppet_agent_end_run/puppet_agent_end_run.rb

```diff
@@ -0,0 +1,33 @@
+Puppet::Type.type(:puppet_agent_end_run).provide :puppet_agent_end_run do
+  def end_run
+    false
+  end
+
+  def needs_upgrade
```

Comment:
you could rename this to `needs_upgrade?` as it is the convention for methods that return a boolean